src/GRANULAR/fix_temp_integrate.cpp

```diff
@@ -0,0 +1,136 @@
+// clang-format off
+/* -*- c++ -*- ----------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   LAMMPS development team: developers@lammps.org
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "fix_temp_integrate.h"
+
+#include "atom.h"
+#include "error.h"
+#include "force.h"
+#include "memory.h"
+#include "respa.h"
+#include "update.h"
+
+using namespace LAMMPS_NS;
+using namespace FixConst;
+
+enum {NONE, CONSTANT, TYPE};
+
+/* ---------------------------------------------------------------------- */
+
+FixTempIntegrate::FixTempIntegrate(LAMMPS *lmp, int narg, char **arg) :
+  Fix(lmp, narg, arg)
+{
+  if (narg < 4) error->all(FLERR,"Illegal fix command");
+
+  cp_style = NONE;
+
+  int ntypes = atom->ntypes;
+  if (strcmp(arg[3],"constant") == 0) {
+    if (narg != 5) error->all(FLERR,"Illegal fix command");
+    cp_style = CONSTANT;
+    cp = utils::numeric(FLERR,arg[4],false,lmp);
+    if (cp < 0.0) error->all(FLERR,"Illegal fix command");
+  } else if (strcmp(arg[3],"type") == 0) {
+    if (narg != 4 + ntypes) error->all(FLERR,"Illegal fix command");
+    cp_style = TYPE;
+    memory->create(cp_type,ntypes+1,"fix/temp/integrate:cp_type");
+    for (int i = 1; i <= ntypes; i++) {
+      cp_type[i] = utils::numeric(FLERR,arg[3+i],false,lmp);
+      if (cp_type[i] < 0.0) error->all(FLERR,"Illegal fix command");
+    }
+  } else {
+    error->all(FLERR,"Illegal fix command");
+  }
+
+  if (cp_style == NONE)
+    error->all(FLERR, "Must specify specific heat in fix temp/integrate");
+  dynamic_group_allow = 1;
+  time_integrate = 1;
```

Comment:
I have some formal concerns about this fix.
- It is flagged as doing time integration, but particle positions are unchanged. If a position update is done with a different fix, there will be warnings about atoms being time integrated twice. But that is not really true in this case.
- All fixes that have names starting with "temp/" are thermostats, so having the "temp/integrate" name is in conflict with that convention
- For time integration fixes that are specific to some atoms styles, we have a convention to use something like fix nve/atom-style (see. fix nve/sphere, fix nvt/asphere, fix nve/bpm/sphere etc.) so this fix's name should end in sphere/temp.

So something like `fix heat/flow/sphere/temp` or `fix heat/transport/sphere/temp` could be more in line with the existing conventions.

Reply:
I'll swap it to one of those names and fix the incorrect flag. Thanks for pointing those out!